### PR TITLE
chore: add SAST scanning

### PR DESCRIPTION
**Pull request by justinabrahms**
Refs #84

Signed-off-by: Justin Abrahms <justin@abrah.ms>

### CODE REVIEW

File: .github/workflows/static-code-scanning.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    #        ┌───────────── minute (0 - 59)
+    #        │  ┌───────────── hour (0 - 23)
+    #        │  │ ┌───────────── day of the month (1 - 31)
+    #        │  │ │ ┌───────────── month (1 - 12 or JAN-DEC)
+    #        │  │ │ │ ┌───────────── day of the week (0 - 6 or SUN-SAT)
+    #        │  │ │ │ │
+    #        │  │ │ │ │
+    #        │  │ │ │ │
+    #        *  * * * *
+    - cron: '30 1 * * 1'
```
Comment (toddbaert) [inline .github/workflows/static-code-scanning.yaml:20]: woah.

**Review by toddbaert — APPROVED** [review]
I wanted to make sure we were OK to use this, and I found this.

> CodeQL is free for research and open source.

https://codeql.github.com/

I'm not sure if there's any specific requirements we need to satisfy.

LGTM

**Comment by justinabrahms** [comment]
I'm currently experimenting with how to get the build time for this step lower. If I can't, I'll probably kick it off on a schedule.

**Comment by justinabrahms** [comment]
> GitHub CodeQL can only be used on codebases that are released under an OSI-approved open source license, or to perform academic research, or to generate CodeQL databases for or during automated analysis, continuous integration (CI) or continuous delivery (CD) in the following cases: (1) on any Open Source Codebase hosted and maintained on GitHub.com, and (2) to test CodeQL queries you have released under an OSI-approved open source software license. It can't be used for automated analysis, continuous integration or continuous delivery, whether as part of normal software engineering processes or otherwise, except in the express cases set forth herein. For these uses, [contact the sales team.](https://enterprise.github.com/contact)

Seems like we're pretty squarely in the good here. Interesting terms tho!

**Review by toddbaert — APPROVED** [review]

**Review by toddbaert — APPROVED** [review]

**Pull request merged**